hudi-common/src/test/java/org/apache/hudi/common/util/TestSerializationUtils.java

```diff
@@ -80,6 +86,30 @@ public void testClassFullyQualifiedNameSerialization() throws IOException {
     assertEquals(ByteBuffer.wrap(firstBytes), ByteBuffer.wrap(secondBytes));
   }
 
+  @Test
+  public void testSerHoodieMetadataPayload() throws IOException {
```

Comment:
Let's move this test to hudi-spark module to make sure it's being run against every Spark version